nestkernel/connection_creator_impl.h

```diff
@@ -438,7 +438,7 @@ ConnectionCreator::fixed_indegree_( Layer< D >& source,
         std::vector< bool > is_selected( positions.size() );
 
         // Draw `target_number_connections` sources
-        for ( int i = 0; i < ( int ) target_number_connections; ++i )
+        for ( int i = 0; i < static_cast< int >( target_number_connections ); ++i )
```

Comment:
Could you create a follow-up issue to revise the horrible coding pattern here? `target_number_connections` is an `unsigned long` above and should just be counted down in a `while` loop here. Instead we cast to `int` so that we can do `--i` to counteract the loops `++i` if a connection is not created.

This PR should focus just on the casts, thus the suggestion to follow up in a separate issue. Applies also to subsequent cases.

Reply:
Somehow completely missed your comment. I created a follow-up PR cleaning up this bad code design (#2597 ).